dags/htan-validate-dag.py

```diff
@@ -19,32 +20,21 @@
 )
 def htan_nf_dcqc_dag():
     @task()
-    def get_synapse_input_file() -> str:
+    def stage_input_synapse_to_s3() -> Path:
         """
-        Gets synapse file (input csv) from synapse and saves it in buffer.
-        Passes along dict object with the path to the downloaded file and the name of the file.
+        Gets synapse file (input csv) from synapse and saves it in temp_dir. Then stages file in S3
+        Passes along path to S3 location of file and cleans up temp_dir.
 
         Returns:
-            str: String containing the path to the downloaded file and the name of the file.
+            s3_uri (str): Path to S3 bucket location of file
         """
-        # simple param passing from run with config - sets my test file as default for now
-        syn_id = get_current_context()["params"].get("syn_id") or "syn50919899"
+        syn_id = get_current_context()["params"].get("syn_id", "syn50919899")
         syn_token = Variable.get("SYNAPSE_AUTH_TOKEN")
         syn = synapseclient.login(authToken=syn_token)
-        file_path = syn.get(syn_id).path
-        return file_path
-
-    @task()
-    def stage_input_in_s3(file_path: str) -> str:
-        """
-        Uploads file to Nextflow Tower S3 bucket, returns string path to file
-        Args:
-            file_path (str): String containing the path to the downloaded file to be uploaded to S3
-
-        Returns:
-            s3_uri (str): Path to S3 bucket location of file
-        """
+        temp_dir = TemporaryDirectory()
+        file_path = Path(syn.get(syn_id, downloadLocation=temp_dir.name).path)
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** I would avoid calling more than one function in a line for readability. Even if you only use a variable once, it helps understand the right-hand side of the assignment without needing a comment. 

```suggestion
        syn_file = syn.get(syn_id, downloadLocation=temp_dir.name)
        file_path = Path(syn_file.path)
```